### ISSUE TITLE
[BOUNTY] [FEATURE] Implement unbounded horizontal scroll

### ISSUE THREAD

**Issue by roha99** [fleaflet/flutter_map#1582]
### What do you want implemented?

i'm absolutely in love with flutter_map ;) thanks a lot for this awesome package !
however an option like [worldCopyJump](https://leafletjs.com/reference.html#map-worldcopyjump) in [leaflet](https://leafletjs.com) would be quite essential to my use case :)
i've double-checked and it's the default behavior of most major mapping apps (e.g. google and apple)
it would be cool to recreate such a user experience with flutter_map ;)
#1201

### Severity

Obtrusive: No workarounds are available, and this is essential to me

## Bounty

This issue is eligible for a bounty!

### Bounty Criteria

To be eligible to recieve the bounty (as outlined in the meta information below), the solution must:
* implement seamless infinite horizontal scrolling of the map
* allow other layers to use this functionality - eg. by creating a framework or specification to allow for this capability to be implemented in other layers
* implement infinite tiling and scrolling on the standard `TileLayer`
* (optional, but appriciated (no additional bounty)) implement infinite repetition of `Marker`s in the `MarkerLayer`

An example of another open source mapping library that offers infinite/unbounded horizontal scrolling is leaflet.js.

The decision to award the bounty is entirely at the discretion of the maintainer team.

### Bounty Meta Information

flutter_map uses [Polar.sh](https://polar.sh/fleaflet) to provide a bounty on this issue! Contributors who resolve this issue using the criteria listed above will receive the full amount pledged below, after all fees have been paid.

The flutter_map team funds issues we want to see fixed but are relatively low priority for us, and we do this using your donations! When you [donate](https://docs.fleaflet.dev/supporters#support-us), 15% goes to the OpenStreetMap Foundation, and the rest is held as donations, some of which we fund these issues with!

Of course, we appriciate any size donation you can make: it supports us directly, and allows us to fund these issues, but if this issue is particularly important to you, you can pledge money directly to this issue using the button below.
<br><br>

<a href="https://polar.sh/fleaflet/flutter_map/issues/1582">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/fleaflet/flutter_map/issues/1582/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/fleaflet/flutter_map/issues/1582/pledge.svg">
</picture>
</a>

**Comment by JaffaKetchup** [comment]
Hi @r0b3rth4ns3n, thanks for your interest!
At the moment, I think this depends on #1452, which is marked as P4 due to its low-ish demand and due to the fact it won't immediately lead to any new functionality.

**Comment by ibrierley** [comment]
Hmm I'm not entirely sure it does, if I'm reading that one right. The LatLngs would still be the same wouldn't they ?

My thinking is probably more like that a custom crs would need to be used, and the infinite flag set to true, but I doubt that code fully works, but it would probably be my first port of call when examining...

However,as ever the devil is in the detail. Let's say you have a marker in London and stretch the map and London is displayed twice (or is there a limit to only display one full map) I.e Google seems to have a fixed limit so is infinite, but not duplicated, i.e you can't have 2 Londons displayed.. leaflet.js I think you can have 2 Londons displayed (i.e at https://leafletjs.com/examples/quick-start/ ) but then it only displays one set of markers, so as you scroll horizontally you actually lose your marker. Or should it be more like https://fluttermap.dabbles.info/geoslicer/#/ which isn't infinite, but will duplicate as it stretches out...

**Comment by JaffaKetchup** [comment]
I guess it depends on how we treat it.
If we keep the current `LatLng`, we limit to +/- 180deg. I guess this means any taps in the non-standard 'region' will be redirected to taps in the standard region? Ie. Trying to place a new marker at the tap position would place the marker in the corresponding tap position on the standard region.

**Comment by r0b3rth4ns3n** [comment]
in case someone else is looking for this feature right now
both #1452 and #468 mention possible albeit undesirable workarounds

**Comment by ibrierley** [comment]
Yeah, this is why it's one of those that sounds simple and fine in theory until its broken down. 

Fwiw I feel like Googles feels the most intuitive. I.e limit the size to only one world, but allow scrolling to the same map/markers wrapped around. Leaflet feels incorrect, allows infinite scrolling, but then you're kind of on a separate map not showing the existing markers which feels confusing.

**Comment by r0b3rth4ns3n** [comment]
i agree with @ibrierley
the user experience of google maps seems most intuitive

**Comment by r0b3rth4ns3n** [comment]
imho hitting a wall at "the edge of the world" as with the current behavior of flutter_map seems most undesirable in terms of user experience and plays into flat earthers hands ;)

**Comment by jbdtky** [comment]
I believe it's a very important feature for any person using a map in a app. It's a feature that definitely need ASAP. So happy to contribute if needed.

**Comment by JaffaKetchup** [comment]
We're very happy to accept PRs, but it's likely going to be a big job!

**Comment by jbdtky** [comment]
I need it for my product: https://www.dokoni.app. Or maybe should I use something else, any recommendation?

**Comment by JaffaKetchup** [comment]
Personally, I'm a massive supporter of OSS and not locking in to massive corporations. Aware I'm using GitHub 😂

On a serious note, the only other real alternatives are mapbox_gl and Google. But you've probably looked into those already. For flexibility and ecosystem, FM really is the leader atm.

**Comment by noforeignland** [comment]
Anyu update on this? I'm looking at the viability of using FM in an app that allows users to pan around in the Pacific and it's really not intuitive for them to have to scroll the "wrong" way around the world when they reach the anti-meridian. Thx.

**Comment by ReinisSprogis** [comment]
Something like this? 
https://github.com/fleaflet/flutter_map/assets/69913791/f1574401-fab1-4575-a024-6f31835e7b7c

**Comment by JaffaKetchup** [comment]
Hey @ReinisSprogis,
Yep, this is pretty close to what we're looking for. However, for it to be fully seamless, the flash would need to be removed. Is that possible?
If you're actively working on this, please open a draft PR and I'll assign you to this issue.

**Comment by monsieurtanuki** [comment]
@JaffaKetchup For the record I've just managed to scroll beyond the end of the world, and to remove the undesired flash side-effect.
Needs some refactoring, though.
Am I supposed to PR a draft now?

**Comment by JaffaKetchup** [comment]
Yes, a draft PR would be great! I'll assign you to this issue.

**Comment by monsieurtanuki** [comment]
Thank you @JaffaKetchup!
I had the wrong impression I solved it a couple of days ago.
But eventually I've found and coded today the solution against the flash effect.
Would probably be ready for a first lousy PR tomorrow.
